source/log4net-loggly/LogglyClient.cs

```diff
@@ -1,52 +1,111 @@
 using System;
+using System.Collections.Generic;
 using System.Net;
 using System.Text;
+using System.Linq;
 
 namespace log4net.loggly
 {
 	public class LogglyClient : ILogglyClient
 	{
+		static bool isValidToken = true;
+		public static void setValidInvalidFlag(bool flag)
+		{
+		   isValidToken = flag;
+		}
+
 		public virtual void Send(ILogglyAppenderConfig config, string message)
 		{
-            int maxRetryAllowed = 5;
-            int totalRetries = 0;
-
-            string _tag = config.Tag;
-            
-            //keeping userAgent backward compatible
-            if (!string.IsNullOrWhiteSpace(config.UserAgent))
-            {
-                _tag = _tag + "," + config.UserAgent;
-            }
-
-            while (totalRetries < maxRetryAllowed)
-            {
-                totalRetries++;
-                try
-                {
-			        var bytes = Encoding.UTF8.GetBytes(message);
-                    var webRequest = CreateWebRequest(config, _tag);
-
-                    using (var dataStream = webRequest.GetRequestStream())
-			        {
-				        dataStream.Write(bytes, 0, bytes.Length);
-				        dataStream.Flush();
-				        dataStream.Close();
-			        }
-            
-                    var webResponse = webRequest.GetResponse();
-                    webResponse.Close();
-                    break;
-                }
-                catch { }
-            }
+			int maxRetryAllowed = 5;
+			int totalRetries = 0;
+
+			string _tag = config.Tag;
+			bool isBulk = config.LogMode.Contains("bulk");
+
+			List<string> messageBulk = new List<string>();
+			//keeping userAgent backward compatible
+			if (!string.IsNullOrWhiteSpace(config.UserAgent))
+			{
+				_tag = _tag + "," + config.UserAgent;
+			}
+
+			while (isValidToken && totalRetries < maxRetryAllowed)
+			{
+				totalRetries++;
+				try
+				{
+					var bytes = Encoding.UTF8.GetBytes(message);
+					var webRequest = CreateWebRequest(config, _tag);  
+					
+						using (var dataStream = webRequest.GetRequestStream())
+						{
+							dataStream.Write(bytes, 0, bytes.Length);
+							dataStream.Flush();
+							dataStream.Close();
+						}
+
+						var webResponse = webRequest.GetResponse();
+						webResponse.Close();
+						break;                    
+				}
+
+				catch (WebException e) {
+					var response = (HttpWebResponse)e.Response;
+					if (response != null) 
+					{
+						if (response.StatusCode == HttpStatusCode.Forbidden)  //Check for bad token
+						{
+							setValidInvalidFlag(false);
+						}
+						if (totalRetries == 1) Console.WriteLine("Loggly error: {0}", e.Message);
+					}
+
+					else if (totalRetries == 1)

```

Comment:
This logic of else if (totalRetries == 1) is also very confusing since this has no relation with above if (response != null). We should always have if and else if in conjunction. I understand what you are trying to do here. You are basically saying if there is exception and the it is not first time lets put it in buffer. I think we can make code such that it is more easy to understand.